lib/SIL/TypeLowering.cpp

```diff
@@ -2520,3 +2520,69 @@ TypeConverter::getContextBoxTypeForCapture(ValueDecl *captured,
   
   return boxType;
 }
+
+size_t Lowering::countNumberOfFields(SILModule &Module, SILType Ty) {
+  size_t fieldsCount = 0;
+  CanType canType = Ty.getSwiftRValueType();
+
+  std::function<bool(Type)> traverseAndCount = [&](Type type) -> bool {
+    if (auto *structDecl = type->getStructOrBoundGenericStruct()) {
+      for (auto prop : structDecl->getStoredProperties()) {
+        if (!prop->hasType()) {
+          continue;
+        }
+        auto propTy = prop->getType();
```

Comment:
Use getInterfaceType() instead. At the SIL level you will only have fully type checked declarations, so it will never return null